### PR TITLE
awx aint the only user in town

### PR DESCRIPTION
**Pull request by kdelee**
##### SUMMARY

Building off of https://github.com/ansible/awx/pull/10918 , I wan to see if this makes me able to run a project update

### CODE REVIEW

File: awx/main/models/credential/injectors.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def gce(cred, env, private_data_dir):
     f = os.fdopen(handle, 'w')
     json.dump(json_cred, f, indent=2)
     f.close()
-    os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
+    os.chmod(path, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP)
```
Comment (AlanCoding) [inline awx/main/models/credential/injectors.py:34]: Seeing this repeated so many times, it makes it seem obvious that it should become a shared constant, and an easy location for this is `awx.main.constants`. We might have multiple variety of file permissions for executable files, and some others.

File: awx/main/models/credential/__init__.py [hunk 2]
```diff
@@ -539,7 +539,7 @@ def build_extra_vars_file(vars, private_dir):
                 f = os.fdopen(handle, 'w')
                 f.write(safe_dump(vars))
                 f.close()
-                os.chmod(path, stat.S_IRUSR)
+                os.chmod(path, stat.S_IRUSR | stat.S_IRGRP)
```
Comment (AlanCoding) [inline awx/main/models/credential/__init__.py:542]: I cannot articulate why the extra vars file should not be writable when the others are.

File: awx/main/models/credential/injectors.py (hunk 1 above)
Comment (kdelee) [reply awx/main/models/credential/injectors.py:34]: +1

File: awx/main/tasks.py [hunk 3]
```diff
@@ -966,6 +967,10 @@ def build_private_data_files(self, instance, private_data_dir):
         private_data = self.build_private_data(instance, private_data_dir)
         private_data_files = {'credentials': {}}
         if private_data is not None:
+            artifact_dir = os.path.join(private_data_dir, 'artifacts', str(self.instance.id))
+            if not os.path.exists(artifact_dir):
+                os.makedirs(artifact_dir)
+                os.chmod(artifact_dir, stat.S_IRUSR | stat.S_IXUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IXGRP | stat.S_IWGRP)
```
Comment (AlanCoding) [inline awx/main/tasks.py:973]: Why did you have to move artifact dir handling and change the pattern for it?

**Comment by kdelee** [comment]
Abandoning this approach

**Pull request closed without merging**